oktaawscli/okta_auth.py

```diff
@@ -142,6 +154,40 @@ def verify_single_factor(self, factor, state_token):
                         exit(1)
                     else:
                         time.sleep(0.5)
+
+            if factor['factorType'] == 'u2f':
+                devices = u2f.list_devices()
+                if len(devices) == 0:
+                    print("No U2F device found")
+                    exit(1)
+
+                challenge = dict()
+                challenge['appId'] = resp_json['_embedded']['factor']['profile']['appId']
+                challenge['version'] = resp_json['_embedded']['factor']['profile']['version']
+                challenge['keyHandle'] = resp_json['_embedded']['factor']['profile']['credentialId']
+                challenge['challenge'] = resp_json['_embedded']['factor']['_embedded']['challenge']['nonce']
+                auth_response = None
+                while not auth_response:
+                    for device in devices:
+                        with device as dev:
+                            try:
+                                auth_response = u2f.authenticate(dev, challenge, resp_json['_embedded']['factor']['profile']['appId'] )
+                                req_data.update(auth_response)
+                                resp = requests.post(resp_json['_links']['next']['href'], json=req_data)
+                                resp_json = resp.json()
+                                if resp_json['status'] == 'SUCCESS':
+                                    return resp_json['sessionToken']
+                                elif resp_json['factorResult'] == 'TIMEOUT':
+                                    print("Verification timed out")
```

Comment:
We should have any output to the console running through the logger, instead of `print()`.
```suggestion
                                    self.logger.warning("Verification timed out")
```

Reply:
done